src/ctim/schemas/common.clj

```diff
@@ -37,6 +31,54 @@
 
 (def default-tlp "green")
 
+(s/defschema BaseEntity
+  {;; :id and :idref must be implemented exclusively
+   :id ID
+   :type s/Str
+   :schema_version (describe s/Str "CTIM schema version for this entity")
+   (s/optional-key :uri) URI
+   (s/optional-key :revision) s/Int
+   (s/optional-key :external_ids) [s/Str]
+   (s/optional-key :timestamp) Time
+   (s/optional-key :language) s/Str
+   (s/optional-key :tlp) TLP
+   })
+
+(s/defschema NewBaseEntity
+  "Base for New Entities, optionalizes ID and type and schema_version"
+  (st/merge
+   BaseEntity
+   (st/optional-keys
+    {:id ID
+     :type (describe s/Str "A valid entity type identifer")
+     :schema_version (describe s/Str "CTIM schema version for this entity")})))
```

Comment:
We need to allow the client to specify that, and we need to accept all compatible versions -- and that should be determined via semantic versioning comparison of the schema versions.  AKA, if the first two numbers match.  This way, as we have different services generating data, we can know what needs transmogrifying (see Calvin and Hobbes) on import, or to clean up later.

Accepting the schema_version means we can set our own policies for wether we translate things on the fly, or flatly reject them if it's not a supported schema.

Since our ES schema is dynamic: strict -- we actually are tightly tied to the full version number, and are hence very sensitive to deviations in schema even if they are forward compatible (such as adding new fields).  I think we may want to change our dynamic mapping to be "allow but don't index" eventually.  I like keeping it strict now tho, since that helps us identify boogz.